.buildkite/pipeline.yml

```
@@ -79,21 +102,19 @@ steps:
   - label: "Java Integration Tests"
     key: "java_integration_tests"
     command: 
-      - "echo 'Java Integration Tests step is disabled, awaiting fix for cluster tear down.'"
-        #- 'buildkite-agent artifact download "build/*" .'
-        #- "mkdir -p ~/.ssh"
-        #- "ssh-keyscan github.com >> ~/.ssh/known_hosts"
-        #- '.buildkite/scripts/steps/run_es_tests.sh || (cd ../elasticsearch && find x-pack -name logs | xargs tar cvzf logs.tgz && buildkite-agent artifact upload logs.tgz && false)'
-    depends_on: "build_test_linux-x86_64-RelWithDebInfo"
+      - "sudo yum -y install java-17-amazon-corretto-devel"
+      - "java -version"
+      - 'buildkite-agent artifact download "build/*" . --step build_test_linux-aarch64-RelWithDebInfo'
```

Would this work?

```suggestion
      - 'buildkite-agent artifact download "build/*linux-aarch64*" . --step build_test_linux-aarch64-RelWithDebInfo'
```

If it would then we'd avoid downloading pointless stuff, like artifacts for Mac and Windows.

---

I don't think this is necessary as that is what the `--step` parameter is doing (see https://buildkite.com/docs/agent/v3/cli-artifact#downloading-artifacts-example).

Checking the BuildKite logs I can confirm that it is only downloading artifacts uploaded from the `build_test_linux-aarch64-RelWithDebInfo` step:

```
  | 2023-01-19 15:02:22 INFO   Searching for artifacts: "build/*" within step: "build_test_linux-aarch64-RelWithDebInfo" 
  | 2023-01-19 15:02:22 INFO   Found 2 artifacts. Starting to download to: /opt/buildkite-agent/builds/bk-agent-elastic-aws-1674140483898757664/elastic/ml-cpp
  | 2023-01-19 15:02:24 INFO   Successfully downloaded "build/distributions/ml-cpp-8.7.0-SNAPSHOT-linux-aarch64.zip" 44593845 bytes
  | 2023-01-19 15:02:40 INFO   Successfully downloaded "build/distributions/ml-cpp-8.7.0-SNAPSHOT-debug-linux-aarch64.zip" 838695567 bytes
```

---

Great, thanks for confirming